aws-ts-pulumi-webhooks/index.ts

```diff
@@ -0,0 +1,73 @@
+import * as pulumi from "@pulumi/pulumi";
+import * as cloud from "@pulumi/cloud";
+import * as cloudAws from "@pulumi/cloud-aws";
+
+import * as crypto from "crypto";
+
+import * as slack from "@slack/client";
+
+const config = new pulumi.Config();
+
+const stackConfig = {
+    // Webhook secret used to authenticate messages. Must match the value on the
+    // webhook's settings.
+    sharedSecret: config.get("sharedSecret"),
+
+    slackToken: config.require("slackToken"),
+    slackChannel: config.require("slackChannel"),
+};
+
+// Just logs information aincomming webhook request.
+async function logRequest(req: cloud.Request, _: cloud.Response, next: () => void) {
+    const webhookID = req.headers["pulumi-webhook-id"] as string;
+    const webhookKind = req.headers["pulumi-webhook-kind"] as string;
+    console.log(`Received webhook from Pulumi ${webhookID} [${webhookKind}]`);
+    next();
+}
+
+// Webhooks can optionally be configured with a shared secret, so that webhook handlers like this app can authenticate
+// message integrity. Rejects any incomming requests that don't have a valid "pulumi-webhook-signature" header.
+async function authenticateRequest(req: cloud.Request, res: cloud.Response, next: () => void) {
+    const webhookSig = req.headers["pulumi-webhook-signature"] as string;
+    if (!stackConfig.sharedSecret || !webhookSig) {
+        next();
+        return;
+    }
+
+    const payload = <string>req.body.toString();
+    const hmacAlg = crypto.createHmac("sha256", stackConfig.sharedSecret);
+    const hmac = hmacAlg.update(payload).digest("hex");
+
+    const result = crypto.timingSafeEqual(Buffer.from(webhookSig), Buffer.from(hmac));
+    if (!result) {
+        console.log(`Mismatch between expected signature and HMAC: '${webhookSig}' vs. '${hmac}'.`);
+        res.status(400).end("Unable to authenticate message: Mismatch between signature and HMAC");
+        return;
+    }
+    next();
+}
+
+const webhookHandler = new cloudAws.HttpEndpoint("pulumi-webhook-handler");
```

Comment:
May as well just use `cloud` here since you ended up needing to import it - you can then remove the import of `cloudAws`.

Reply:
SGTM